### PR TITLE
[UPDATE] Removed the secondary MQTT topic.

### PR DESCRIPTION
**Pull request by Jasperrr91**
## Short Description:
Fixes a bug where data gets parsed before actually checking if the data is acquired from the correct channel.

Edit: I have also removed the second topic (#pgomapcatch). The map-chat now also uses the #pgo channel to display pokemon on the map.

## Fixes/Resolves/Closes (please use correct syntax):
-

### CODE REVIEW

File: pokemongo_bot/event_handlers/social_handler.py [hunk 1]
```diff
@@ -20,11 +20,15 @@ def __init__(self, bot, clientid=None):
         #print "rc: "+str(rc)
     def mqtt_on_message(self, mqttc, obj, msg):
         #msg.topic+" "+str(msg.qos)+" "+str(msg.payload)
+        print msg.topic
```
Comment (brantje) [inline pokemongo_bot/event_handlers/social_handler.py]: Can you remove this print?

**Comment by solderzzc** [comment]
You need change the web as well.

**Comment by Jasperrr91** [comment]
We use an old commit of the web without MQTT right? The map-chat is updated with the 2nd commit.

**Comment by solderzzc** [comment]
we should not release counter_id to the web, that's why I use two channel.

**Comment by Jasperrr91** [comment]
@solderzzc both channels send out the encounter_id. The only thing the JSON sends that the other channel does not, is the spawn_point_id. (Which we can remove from the JSON if needed).

**Comment by BreezeRo** [comment]
@Jasperrr91 he did not say it does not currently send out encounter_id, he's saying it shouldn't - and I will fix that on the account of this.

We also do not need to broadcast to a second channel, it's fine how it is though what could be done is a user controlled variable to set broadcasting to a different broker.

**Comment by Jasperrr91** [comment]
Don't close this then lol. This can be merged without problem. As this removes the second channel and adds stability as it checks for the correct channel before parsing.

**Comment by BreezeRo** [comment]
Please open a clean PR

**Comment by Jasperrr91** [comment]
What's not clean about it?

**Comment by BreezeRo** [comment]
The channels do not reflect channels officially sponsored by default by the project, please see my post above for more information.

**Comment by Jasperrr91** [comment]
Yes I have read that. What does this have to do with this PR? It improves the current code as is.

If you want me to make the broker/channel variable. I'll do that in a sec. But this is fine as is.

**Comment by BreezeRo** [comment]
Again, this channel "pgo" is not officially sponsored by the project - we do not use it. If you want to push this through I kindly ask that you make a clean PR without use of these channels.

**Comment by Jasperrr91** [comment]
@ro can you please tell me which channel is officially sponsored by the project then?

As the current dev codebase does in fact use "pgo":
```
self.mqttc.publish("pgomapcatch/all/catchable/"+str(data['pokemon_id']), str(data['latitude'])+","+str(data['longitude'])+","+str(data['encounter_id'])+","+str(data['pokemon_id'])+","+str(data['expiration_timestamp_ms'])+","+str(data['pokemon_name']))
json_data = json.dumps(data)
self.mqttc.publish("pgo/all/catchable/"+str(data['pokemon_id']), json_data)
```

Even more so, it is the channel used by the bot to receive pokemon: `self._mqttc.subscribe("pgo/#", 0)`

**Comment by BreezeRo** [comment]
I stand corrected, must of changed very very recently, as a matter of hours. This was not the case.

`@solderzzc both channels send out the encounter_id. The only thing the JSON sends that the other channel does not, is the spawn_point_id. (Which we can remove from the JSON if needed).`

No, both channels do NOT send out the encounter_id.

>                 self.mqttc.publish("pgomapcatch/all/catchable/"+str(data['pokemon_id']), str(data['latitude'])+","+str(data['longitude'])+","+str(data['encounter_id'])+","+str(data['pokemon_id'])+","+str(data['expiration_timestamp_ms'])+","+str(data['pokemon_name']))
                json_data = json.dumps(data)
                self.mqttc.publish("pgo/all/catchable/"+str(data['pokemon_id']), json_data)

**Comment by Jasperrr91** [comment]
No problem. They do both send out the encounter_id 😛 

First publish sends out `data['encounter_id']`. Then the entire data object is converted to JSON and then published to the second channel.

**Comment by BreezeRo** [comment]
I'm not getting that in code, but I've been wrong in this PR before lol so please, if you can show me where the encounter_id is sent through both topics/channels I'd appreciate it.

**Comment by Jasperrr91** [comment]
> self.mqttc.publish("pgomapcatch/all/catchable/"+str(data['pokemon_id']), str(data['latitude'])+","+str(data['longitude'])+","+**str(data['encounter_id'])**+","+str(data['pokemon_id'])+","+str(data['expiration_timestamp_ms'])+","+str(data['pokemon_name']))
Encounter_id in the data object is sent to the `pgomapcatch/all/catchable/` channel.
           

> **json_data** = json.dumps(**data**)
> self.mqttc.publish("pgo/all/catchable/"+str(data['pokemon_id']), **json_data**)
Entire data object (including the encounter_id contained in it) is sent to the `pgo/dall/catchable/` channel.

**Comment by BreezeRo** [comment]
I'm wondering you why removed the # that indicates these lines are commented out and not compiled?

**Comment by Jasperrr91** [comment]
I did no such thing. Haven't changed any python comments.

**Comment by zeronivram** [comment]
hahaha

**Comment by BreezeRo** [comment]
@Jasperrr91 you agree that since only the pokemon_id is extracted from the array, that no - the entire object is NOT sent to mqtt? If this is true, then we can also agree that again, no, the encounter_id is NOT sent to both channels.

The entire array is NOT sent.

**Comment by Jasperrr91** [comment]
@BreezeRo no I don't agree. The pokemon_id is extracted from the array to construct the topic. Then, after the `,` the `json_data` is sent. And the `json_data` contains the same attributes as the  data object.

So yes, the entire array is indeed sent.

**Comment by BreezeRo** [comment]
I stand corrected, the 3 lines you pasted above are actually one line that resemble the previous two (on dev branch atleast) which are in fact commented out.

Sue me :)

Will merge after encounter_id + other ups are cleared

**Comment by Jasperrr91** [comment]
As proof, I've done a console.log of both channels.  For #pgo the console log is as follows:
```
var pokemon = JSON.parse(payload);
console.log(pokemon);
```

For #pgomapcatch it's as follows:
```
var pLoadR = payload.toString();
console.log(pLoadR);
```

Result:
console_log

The slight discrepancy between the encounter_ids for each channel, is caused by float -> integer correction:
int_correction

**Comment by BreezeRo** [comment]
Which goes back to what I said in my previous post

"he did not say it does not currently send out encounter_id, **he's saying it shouldn't**"

**Comment by Jasperrr91** [comment]
Yes, which brings me back to a point I mentioned before:
>What does this have to do with this PR? It improves the current code as is.

My code is an improvement of the current codebase. I can remove the encounter_id through a future - seperate - PR if you want but you already mentioned before that you would do it.

**Comment by BreezeRo** [comment]
I really don't understand the problem here :hand: , I already said 

"Will merge after encounter_id + other ups are cleared"

**Comment by Jasperrr91** [comment]
Yea and those are cleared up aren't they? 😜

**Comment by BreezeRo** [comment]
I was talking about the parts I need to do

**Comment by Jasperrr91** [comment]
`data.pop('encounter_id', None)`

**Comment by solderzzc** [comment]
We don't merge two channels into one :) 
Close at the time.

**Pull request closed without merging**